components/Typography/tests/unit/MockTraitEnvironment.h

```diff
@@ -0,0 +1,20 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+@interface MockTraitEnvironment : NSObject <UITraitEnvironment>
```

Comment:
Please name this MDCTypographyMockTraitEnvironment since Objective-C has no  namespacing.

Reply:
If/when we share this class in other settings we'll prefix and name it accordingly, but for now the class is only to be used within the context of these tests.

From http://google.github.io/styleguide/objcguide.html#class-names

> Classes and protocols in code shared across multiple applications must have an appropriate prefix (e.g. GTMSendMessage). Prefixes are recommended, but not required, for other classes and protocols.

Reply:
Please fix this name. Another test may also produce a class named "MockTraitEnvironment" and that would cause a symbol collision when the code is linked.  Style guide aside, this is more of a technical issue that has cropped-up in our code base before.

Reply:
Ah. Funzies. Will adjust.

Reply:
Done.